service/controller/clusterapi/v30/key/common.go

```diff
@@ -155,6 +165,24 @@ func RoleARNWorker(getter LabelsGetter, region string, accountID string) string
 	return baseRoleARN(getter, region, accountID, "worker")
 }
 
+// S3ObjectPathTCCP computes the S3 object path to the cloud config uploaded for
+// the TCCP stack.
+//
+//     /version/3.4.0/cloudconfig/v_3_2_5/cluster-al9qy-tccp
+//
+func S3ObjectPathTCCP(getter LabelsGetter) string {
+	return fmt.Sprintf("version/%s/cloudconfig/%s/%s", OperatorVersion(getter), CloudConfigVersion, StackNameTCCP(getter))
+}
+
+// S3ObjectPathTCNP computes the S3 object path to the cloud config uploaded for
+// the TCCP stack.
+//
+//     /version/3.4.0/cloudconfig/v_3_2_5/cluster-al9qy-tcnp-g3j50
```

Comment:
We just use the stack names as identifier for the S3 Objects as this is already a known differentiator.